py3status/request.py

```diff
@@ -63,6 +63,11 @@ def __init__(self, url, params, data, headers, timeout, auth):
             elif isinstance(e, HTTPError):
                 self._status_code = e.code
                 self._error_message = reason
+                # we return an HttpResponse but have no response
+                # so create some 'fake' response data.
+                self._text = ''
+                self._json = None
+                self._headers = []
```

Comment:
Why would you do that? Now it's impossible for modules to handle HTTPError, like redirect loop and stuff. Why not just raise the actual HTTPError instead ?

Reply:
This was to fix occasional errors that were happening whilst keeping existing modules working.  We can probably do some work around improving the situation.

Ultimately the aim is to let modules make simple http/s request without having to do too much work and removing requests as a dependency for many modules that were using it.

We aim for the core functionality of py3status to be as light as possible with preferably no dependencies other than the standard library.  Some modules will need extra dependencies but we shouldn't just for things like a simple http request which is a common need.

Reply:
I think throwing relevant errors instead of making it lie would make it more simple: the request function will have less cases to care of and the module will have more freedom and control over what they are doing. It already throw errors in some case, so it's inconsistent not to throw error in other cases.

Actually I see no point in creating new exceptions that give no information over throwing original `urllib` exception, which is a standard library.

I can tell it from my new module developing experience. If I remember correctly, a few months ago, you were the one to criticize one of my pull request (and you were right) because I wasn't catching enough exceptions. Well, this implementation just makes it harder, because there is no clean way to detect most http error cases.